daffodil-cli/src/main/scala/org/apache/daffodil/Main.scala

```diff
@@ -454,30 +454,37 @@ class CLIConf(arguments: Array[String]) extends scallop.ScallopConf(arguments) {
     footer("""|
               |Run 'daffodil generate <language> --help' for subcommand specific options""".stripMargin)
 
-    object c extends scallop.Subcommand("c") {
-      banner("""|Usage: daffodil generate c -s <schema> [-r <root>]
-                |                           [-c <file>] [-T<tunable>=<value>...]
-                |                           [outdir]
-                |
-                |Generate C code from a DFDL schema to parse or unparse data
-                |
-                |Generate Options:""".stripMargin)
-
-      descr("Generate C code from a DFDL schema")
+    // Takes language by name so we can pass it to scallop.Subcommand and interpolate it
+    // into strings without getting a puzzling java.lang.ClassCastException at runtime (class
+    // scala.collection.mutable.WrappedArray$ofRef cannot be cast to class java.lang.String)
+    class LanguageConf(languageArg: => String) extends scallop.Subcommand(languageArg) {
+      val language = languageArg
+
+      banner(s"""|Usage: daffodil generate $language -s <schema> [-r <root>]
+                 |                              [-c <file>] [-T<tunable>=<value>...]
+                 |                              [outdir]
+                 |
+                 |Generate ${language.toUpperCase} code from a DFDL schema to parse or unparse data
```

Comment:
Thoughts on making `val language = languageArg.toUpperCase` and then just using `language` in the banner/options?

That way if we ever support languages where the language name isn't as simple as the upper cased command then they can override it, e.g.:

```scala
object rust extends LanguageConf("rust") {
  override val language = "Rust"
}

object rust extends LanguageConf("cpp") {
  override val language = "C++"
} 

object rust extends LanguageConf("csharp") {
  override val language = "C#"
}
```

Reply:
I like your suggestion for better extensibility!  I still need to show the original lowercase language name in "daffodil generate c" usage messages but I'll use languageArg for that purpose.

I can't put a timeline on when a new supported language might come out, but I hope it will come eventually.